### PR TITLE
CPLYTM-374 feat(manager): add PluginManager and policy plugin dispensing

### PR DESCRIPTION
**Pull request by jpower432**
Blocked by #7 and #3

### CODE REVIEW

File: framework/config/config.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	store, err := rules.NewMemoryStoreFromComponents(allComponents)
+	if err != nil {
+		return store, titleByID, err
+	}
+	return store, titleByID, nil
+}
+
+// resolveConfiguration returns processed OSCAL Components and a plugin identifier map. This performs most
+// of the logic in ResolveConfiguration, but is broken out to make it easier to test.
```
Comment (qduanmu) [inline framework/config/config.go]: I think these are typos, 'resolveConfiguration' should be 'resolveOptions', and 'ResolveConfiguration' should be 'ResolveOptions' in above comment.

File: framework/config/config.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	return &C2PConfig{
+		PluginDir:            DefaultPluginPath,
+		Logger:               defaultLogger,
+		ComponentDefinitions: make([]oscalTypes.ComponentDefinition, 0),
+	}
+}
+
+// Validate returns an error if C2PConfig has invalid fields.
+func (c *C2PConfig) Validate() error {
+	if c.PluginDir == "" {
```
Comment (qduanmu) [inline framework/config/config.go]: Is it necessary to run strings.TrimSpace on this field value?

File: framework/config/doc.go [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/*
+ Copyright 2024 The OSCAL Compass Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package config defines and processing configuration options for the framework plugin manager.
```
Comment (qduanmu) [inline framework/config/doc.go]: 'processing' should be 'processes'?

File: framework/config/config.go (hunk 1 above)
Comment (jpower432) [reply framework/config/config.go]: Yes. Thank you!

**Review by gvauter — APPROVED** [review]
LGTM

**Review by gvauter — APPROVED** [review]
LGTM

**Pull request merged**